pandas/core/tools/numeric.py

```diff
@@ -142,6 +147,10 @@ def to_numeric(arg, errors="raise", downcast=None):
     else:
         values = arg
 
+    if is_extension_array_dtype(arg) and isinstance(values, NumericArray):
```

Comment:
this really should be part of the above logic. also mask needs to be defined for all cases (can be default to None)

Reply:
> this really should be part of the above logic.

The reason for having it down here is to handle EA dtype Series and array in a single place (and Index when #34159/#37869 go through)

> also mask needs to be defined for all cases (can be default to None)

done